presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -1078,7 +1078,7 @@ public ConnectorTableHandle createTemporaryTable(ConnectorSession session, List<
         validateColumns(storageFormat, columnHandles);
 
         HiveStorageFormat finalStorageFormat = storageFormat;
-        String tableName = PRESTO_TEMPORARY_TABLE_NAME_PREFIX + finalStorageFormat.name() + "_" + randomUUID().toString().replaceAll("-", "_");
+        String tableName = PRESTO_TEMPORARY_TABLE_NAME_PREFIX + finalStorageFormat.name() + "_" + session.getQueryId() + randomUUID().toString().replaceAll("-", "_");
```

Comment:
Presto query strings typically have dashes.  To be safe, consider replacing dashes with underscores, as the random UUID has done.  To make the query obvious, consider surrounding with two underscores.

Reply:
Done.